laplace/baselaplace.py

```diff
@@ -1568,8 +1567,1025 @@ def sample(self, n_samples=100, generator=None):
 
 
 class FunctionalLaplace(BaseLaplace):
-    pass
+    """
+    Applying the GGN (General Gauss Newton) approximation for the Hessian in the Laplace approximation of the posterior
+    turns the underlying probabilistic model from a BNN into a GLM (generalized linear model).
+    This GLM (in the weight space) is equivalent to a GP (in the function space), see
+    [Approximate Inference Turns Deep Networks into Gaussian Processes (Khan et al., 2019)](https://arxiv.org/abs/1906.01930)
+
+    This class implements the (approximate) GP inference through which
+    we obtain the desired quantities (posterior predictive, marginal log-likelihood).
+    See [Improving predictions of Bayesian neural nets via local linearization (Immer et al., 2021)](https://arxiv.org/abs/2008.08400)
+    for more details.
+
+    Note that for `likelihood='classification'`, we approximate \( L_{NN} \\) with a diagonal matrix
+    ( \\( L_{NN} \\) is a block-diagonal matrix, where blocks represent Hessians of per-data-point log-likelihood w.r.t.
+    neural network output \\( f \\), See Appendix [A.2.1](https://arxiv.org/abs/2008.08400) for exact definition). We
+    resort to such an approximation because of the (possible) errors found in Laplace approximation for
+    multiclass GP classification in Chapter 3.5 of [R&W 2006 GP book](http://www.gaussianprocess.org/gpml/),
+    see the question
+    [here](https://stats.stackexchange.com/questions/555183/gaussian-processes-multi-class-laplace-approximation)
+    for more details. Alternatively, one could also resort to *one-vs-one* or *one-vs-rest* implementations
+    for multiclass classification, however, that is not (yet) supported here.
+
+    Parameters
+    ----------
+    M : int
```

Comment:
I prefer a more descriptive name for this parameter. Maybe `num_data`?

Reply:
(It's fine to use `M` internally, but for a user-facing interface, a descriptive name is better)